From 5d265996597a6b0d654bbeda1bc3bae197061de7 Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Mon, 7 Oct 2024 23:43:32 +0100
Subject: [PATCH 2/8] block: partitions: of: assign Device Tree node to
 partition

Assign partition of_node so other drivers are able to identify a
partition by its Device Tree node.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>
---
 block/partitions/of.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/block/partitions/of.c
+++ b/block/partitions/of.c
@@ -48,7 +48,7 @@ static void add_of_partition(struct pars
 	u64 offset = of_read_number(reg, a_cells) / SECTOR_SIZE;
 	u64 size = of_read_number(reg + a_cells, s_cells) / SECTOR_SIZE;
 
-	put_partition(state, slot, offset, size);
+	of_put_partition(state, slot, offset, size, np);
 
 	if (of_property_read_bool(np, "read-only"))
 		state->parts[slot].flags |= ADDPART_FLAG_READONLY;
